ops/framework.py

```diff
@@ -484,6 +484,14 @@ def define_event(cls, event_kind: str, event_type: 'Type[EventBase]'):
 
         event_type: a type of the event to define.
 
+        ops uses a labeling system to track and reconstruct events between hook
+        executions (each time a hook runs, the Juju Agent invokes a fresh
+        instance of ops; there is no ops process that persists on the host
+        between hooks).
+        Having duplicate Python objects creates duplicate labels, which
+        increases the risk of ops running the "wrong" code after
```

Comment:
"the risk" makes it sound nondeterministic. I think @pengale would agree that a better phrasing would be:

overwriting a previously created label means that only the latter code path will be ran when the current event, if it gets deferred, is reemitted. This is usually not what you want, and is error-prone and ambiguous.